viogpu/shared/viogpum.h

```diff
@@ -42,17 +73,126 @@ typedef struct _VIOGPU_DISP_MODE {
 }VIOGPU_DISP_MODE, *PVIOGPU_DISP_MODE;
 #pragma pack()
 
+
+#pragma pack(1)
+typedef struct _VIOGPU_PARAM_REQ {
+    ULONG ParamId;
+    UINT64 Value;
+}VIOGPU_PARAM_REQ;
+#pragma pack()
+
+
+#pragma pack(1)
+typedef struct _VIOGPU_CAPSET_REQ {
+    ULONG CapsetId;
+    ULONG Version;
+    ULONG Size;
+    UCHAR *Capset;
+}VIOGPU_CAPSET_REQ;
+#pragma pack()
+
+
+#pragma pack(1)
+typedef struct _VIOGPU_RES_INFO_REQ {
+    D3DKMT_HANDLE ResHandle;
+    ULONG Id;
+}VIOGPU_RES_INFO_REQ;
+#pragma pack()
+
+
+#pragma pack(1)
+typedef struct _VIOGPU_RES_BUSY_REQ {
+    D3DKMT_HANDLE ResHandle;
+    BOOL Wait;
+    BOOL IsBusy;
+}VIOGPU_RES_BUSY_REQ;
+#pragma pack()
+
+
+#pragma pack(1)
+typedef struct _VIOGPU_CTX_INIT_REQ {   
+    UINT CapsetID;
+}VIOGPU_CTX_INIT_REQ;
+#pragma pack()
+
+
 #pragma pack(1)
 typedef struct  _VIOGPU_ESCAPE{
     USHORT      Type;
     USHORT      DataLength;
     union {
         ULONG Id;
         VIOGPU_DISP_MODE Resolution;
+        VIOGPU_PARAM_REQ Parameter;
+        VIOGPU_CAPSET_REQ Capset;
+
+        VIOGPU_RES_INFO_REQ ResourceInfo;
+        VIOGPU_RES_BUSY_REQ ResourceBusy;
+        
+        VIOGPU_CTX_INIT_REQ CtxInit;
     } DUMMYUNIONNAME;
 } VIOGPU_ESCAPE, *PVIOGPU_ESCAPE;
 #pragma pack()
 
+// ================= CreateResource
+#pragma pack(1)
+typedef struct _VIOGPU_RESOURCE_OPTIONS {
+    ULONG target;
+    ULONG format;
+    ULONG bind;
+    ULONG width;
+    ULONG height;
+    ULONG depth;
+    ULONG array_size;
+    ULONG last_level;
+    ULONG nr_samples;
+    ULONG flags;
+} VIOGPU_RESOURCE_OPTIONS;
+
+
+#pragma pack(1)
+typedef struct _VIOGPU_CREATE_RESOURCE_EXCHANGE {
+    ULONG magic;
+} VIOGPU_CREATE_RESOURCE_EXCHANGE;
+#pragma pack()
+
+#pragma pack(1)
+typedef struct _VIOGPU_CREATE_ALLOCATION_EXCHANGE {
+    VIOGPU_RESOURCE_OPTIONS ResourceOptions;
+    ULONGLONG Size;
+} VIOGPU_CREATE_ALLOCATION_EXCHANGE;
+#pragma pack()
+
+
+// ================= COMMAND BUFFER
+#define VIOGPU_CMD_NOP                          0x0
+#define VIOGPU_CMD_SUBMIT                       0x1 // Submit Command to virgl
+#define VIOGPU_CMD_TRANSFER_TO_HOST             0x2 // Transfer resource to host
+#define VIOGPU_CMD_TRANSFER_FROM_HOST           0x3 // Transfer resource to host
```

Comment:
`Transfer resource to host` -> `Transfer resource from host`

Reply:
Fixed.